tests/test_set_config.py

```diff
@@ -399,3 +394,106 @@ def test_values_needed_f90nml(): #pylint: disable=unused-variable
+
+        out_file = f'{tmp_dir}/test_ouput.yaml'
+        args = ['-i', input_file, '-o', out_file, '--input_file_type', '.yaml']
+
+        set_config.create_config_obj(args)
+
+        expected = config.YAMLConfig(input_file)
+        expected_file = f'{tmp_dir}/test.yaml'
+        expected.dump_file(expected_file)
+
+        assert compare_files(out_file, expected_file)
+
+        with open(out_file, 'r', encoding='utf-8') as output:
+            assert output.read()[-1] == '\n'
+
+def test_output_file_conversion(): #pylint: disable=unused-variable
+    ''' Test that --output_input_type converts config object to desired object type'''
+    input_file = os.path.join(uwtools_file_base, pathlib.Path("fixtures/simple.nml"))
+
+    with tempfile.TemporaryDirectory(dir='.') as tmp_dir:
+
+        out_file = f'{tmp_dir}/test_ouput.cfg'
+        args = ['-i', input_file, '-o', out_file, '--output_file_type', '.nml']
+
+        set_config.create_config_obj(args)
+
+        expected = config.F90Config(input_file)
+        expected_file = f'{tmp_dir}/expected_nml.nml'
+        expected.dump_file(expected_file)
+
+        assert compare_files(expected_file, out_file)
+
+        with open(out_file, 'r', encoding='utf-8') as output:
+            assert output.read()[-1] == '\n'
+
+def test_config_file_conversion(): #pylint: disable=unused-variable
+    ''' Test that --config_input_type converts config object to desired object type'''
+    input_file = os.path.join(uwtools_file_base, pathlib.Path("fixtures/simple2.nml"))
+    config_file = os.path.join(uwtools_file_base, pathlib.Path("fixtures/simple2.ini"))
+
+    with tempfile.TemporaryDirectory(dir='.') as tmp_dir:
+
+        out_file = f'{tmp_dir}/test_config_conversion.nml'
+        args = ['-i', input_file, '-c', config_file, '-o', out_file, '--config_file_type', '.nml']
+
+        set_config.create_config_obj(args)
+
+        expected = config.F90Config(input_file)
+        config_file_obj = config.F90Config(config_file)
+        expected.update_values(config_file_obj)
+        expected_file = f'{tmp_dir}/expected_nml.nml'
+        expected.dump_file(expected_file)
+
+        assert compare_files(expected_file, out_file)
+
+        with open(out_file, 'r', encoding='utf-8') as output:
+            assert output.read()[-1] == '\n'
+
+def test_erroneous_conversion_flags(): #pylint: disable=unused-variable
+    ''' Test that error is thrown when conversion file types are not compatible'''
+
+    with tempfile.TemporaryDirectory(dir='.') as tmp_dir:
+
+        # test --input_file_type
+        input_file = os.path.join(uwtools_file_base, pathlib.Path("fixtures/simple2_nml.cfg"))
+        args = ['-i', input_file, '--input_file_type', ".pdf"]
+
+        with pytest.raises(ValueError) as error:
+            set_config.create_config_obj(args)
+
+            expected = "Set config failure: input base file not compatible\n"
+            actual = str(error.value)
+            assert actual == expected
```

Comment:
I'm not sure these three lines will be reachable because the ValueError should be raised and handled in line 471. You may be able to delete them. Same below.